opensrp-reveal/src/main/java/org/smartregister/reveal/server/FileHTTPServer.java

```diff
@@ -0,0 +1,157 @@
+                Timber.i("Ready for requests on port %d", socket.getLocalPort());
+                while (!isInterrupted()) {
+                    Socket connection = socket.accept();
+                    Timber.i("Accepted a client connection");
+                    new ResponseThread(connection).start();
+                }
+                Timber.i("Server thread interrupted");
+            } catch (IOException e) {
+                Timber.i("Server thread stopped: %s", e.getMessage());
+            }
+        }
+    }
+
+    class ResponseThread extends Thread {
+        final Socket connection;
+
+        ResponseThread(Socket connection) {
+            this.connection = connection;
+        }
+
+        public void run() {
+            try (Socket connection = this.connection) {
+                InputStreamReader reader = new InputStreamReader(connection.getInputStream());
+                String request = new BufferedReader(reader).readLine();
+                Timber.i("Received request: %s", request);
+                if (request == null) {
+                    return;
+                }
+                long start = System.currentTimeMillis();
+                FileHTTPServer.Response response = getResponse(request);
+                if (response == null) {
+                    Timber.i("%s: Style file not found", request);
+                    return;
+                }
+                sendResponse(connection, response);
+                long finish = System.currentTimeMillis();
+                Timber.i("%s: Served %d bytes in %d ms", request, response.data.length, finish - start);
+            } catch (IOException e) {
+                Timber.e(e, "Unable to read request from socket");
+            }
+        }
+
+        protected FileHTTPServer.Response getResponse(String request) {
+            if (request.startsWith("GET /")) {
+                return new FileHTTPServer.Response(styleJson.getBytes(),  "text/plain");
+            }
+            Timber.w("Ignoring request: %s", request);
+            return null;
+        }
+
+        protected void sendResponse(Socket connection, FileHTTPServer.Response response) {
+            String headers = String.format(
+                    Locale.US,
+                    "HTTP/1.0 200\r\n" +
+                            "Content-Type: %s\r\n" +
+                            "Content-Length: %d\r\n" +
+                            "\r\n",
+                    response.contentType,
+                    response.data.length
+            );
+
+            try (OutputStream output = connection.getOutputStream()) {
+                output.write(headers.getBytes());
+                output.write(response.data);
+                output.flush();
+            } catch (IOException e) {
+                Timber.e(e, "Unable to write response to socket");
+            }
+        }
+    }
+
+    public static class Response {
+        byte[] data;
+        String contentType;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=4217760)